src/org/labkey/test/components/domain/DomainPanel.java

```diff
@@ -0,0 +1,154 @@
+package org.labkey.test.components.domain;
+
+import org.labkey.test.Locator;
+import org.labkey.test.components.Component;
+import org.labkey.test.components.WebDriverComponent;
+import org.labkey.test.util.LabKeyExpectedConditions;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+/**
+ * Wraps the functionality of the LabKey ui component defined in domainproperties/CollapsiblePanelHeader.tsx
+ * Subclasses should add panel-specific functionality.
+ */
+public abstract class DomainPanel<EC extends DomainPanel<EC, T>.ElementCache, T extends DomainPanel<?, T>> extends WebDriverComponent<EC>
```

Comment:
Is it possible to make this class a little more generic? If you look at Sample Manager Workflow create a new job, the "Define Job Tasks" is a domain panel but has a few more html elements in the title bar. For example it has text that appears on the right hand side of the heard (ex "1 Task Defined"). It also has an icon next to the panel title.

Reply:
Are you thinking of having methods to fetch the icon class and task count?
I'll look at adding them. Regular domain editors have those elements as well, so that seems reasonable.

Reply:
I also added a class to the title element of the workflow job panes so that these helpers should work with them. I'm not sure how reliable they will be though. It looks like the workflow jobs form doesn't actually use any of the domain components, it just styles itself to look like them.